### DISCUSSION TITLE
Is it recommended to use jemalloc in Node.js ?

### DISCUSSION THREAD

**Discussion by theanarkh** [nodejs/help#4567] [Q&A]
### What is the problem this feature will solve?

I saw some issues earlier about using [jemalloc](https://github.com/nodejs/node/issues?q=jemalloc), has anyone done this in a production environment ? thanks !

### What is the feature you are proposing to solve the problem?

none

**Comment by marco-ippolito** [comment]
Moving this to discussion as it's for gathering feedback

**Comment by laurisvan** [comment]
This answer is a bit late, but we recently switched from musl malloc to jemalloc. We suspected memory fragmentation as the reason for node.js garbage collection failing to a OOM kill. After switching we have not seen such OOM kills yet.

**Reply by laurisvan** [reply]
Update on this one - jemalloc is not a silver bullet here. We have seen OOM kills since that and cannot confirm it certainly would work better. Then again, we are running Alpine containers, and musl malloc therein provenly had several problems. That is not a node.js specific problem, though.

Since Jemalloc is no longer officially maintained (https://jasone.github.io/2025/06/12/jemalloc-postmortem/), a better alternative might be [mimalloc](https://github.com/microsoft/mimalloc) or [tcmalloc](https://google.github.io/tcmalloc/overview.html).
